chain/actors/builtin/registry.go

```diff
@@ -32,31 +38,63 @@ import (
 	"github.com/filecoin-project/lotus/chain/actors"
 )
 
-var _ rtt.VMActor = (*RegistryEntry)(nil)
-
 type RegistryEntry struct {
 	state   cbor.Er
 	code    cid.Cid
-	methods []interface{}
+	methods map[uint64]builtin.MethodMeta
 }
 
 func (r RegistryEntry) State() cbor.Er {
 	return r.state
 }
 
-func (r RegistryEntry) Exports() []interface{} {
+func (r RegistryEntry) Exports() map[uint64]builtin.MethodMeta {
 	return r.methods
 }
 
 func (r RegistryEntry) Code() cid.Cid {
 	return r.code
 }
 
-func MakeRegistry(av actorstypes.Version) []rtt.VMActor {
+func MakeRegistryLegacy(actors []rtt.VMActor) []RegistryEntry {
+	registry := make([]RegistryEntry, 0)
+
+	for _, actor := range actors {
+		methodMap := make(map[uint64]builtin.MethodMeta)
+		for methodNum, method := range actor.Exports() {
+			if method != nil {
+				methodMap[uint64(methodNum)] = makeMethodMeta(method)
+			}
+		}
+		registry = append(registry, RegistryEntry{
+			code:    actor.Code(),
+			methods: methodMap,
+			state:   actor.State(),
+		})
+	}
+
+	return registry
+}
+
+func makeMethodMeta(method interface{}) builtin.MethodMeta {
+	ev := reflect.ValueOf(method)
+	// Extract the method names using reflection. These
+	// method names always match the field names in the
+	// `builtin.Method*` structs (tested in the specs-actors
+	// tests).
+	fnName := runtime.FuncForPC(ev.Pointer()).Name()
```

Comment:
That's mildly scary, but I guess fiine? Would be really nice to have a unit test in lotus making sure we notice when this breaks

Reply:
We do have integ tests that break (cuz they depend on the MethodMeta being setup correctly)...which is also the only reason anyone notices when we break this...